cmake/public/cuda.cmake

```
@@ -60,11 +60,15 @@ find_package(CUDAToolkit REQUIRED)
 
 cmake_policy(POP)
 
+# resolve symlinks: GH-108931
+file(REAL_PATH CUDA_INCLUDE_DIRS _inc_cuda)
+file(REAL_PATH CUDAToolkit_INCLUDE_DIR _inc_ctk)
```

Locally I'm seeing these set to `/home/pbell/git/pytorch/CUDA_INCLUDE_DIRS`
```suggestion
file(REAL_PATH "${CUDA_INCLUDE_DIRS}" _inc_cuda)
file(REAL_PATH "${CUDAToolkit_INCLUDE_DIR}" _inc_ctk)
```